etc/ha_healthchecker/ha_healthchecker.py.j2

```diff
@@ -780,23 +861,59 @@ class Switcher(Base):
                          'name': self.node.name})
 
 
+class LoginHTMLParser(HTMLParser):
+    def handle_startendtag(self, tag, attrs):
+        global LOGIN_AUTH_TOKEN
+        if tag == 'input' and ('name', 'authenticity_token') in attrs:
+            for key, value in attrs:
+                if key == 'value':
+                    LOGIN_AUTH_TOKEN = value
+
+
+class AppUpdateHTMLParser(HTMLParser):
+    token_index = 1
+
+    def handle_startendtag(self, tag, attrs):
+        global APP_UPDATE_AUTH_TOKEN
+        if tag == 'input' and ('name', 'authenticity_token') in attrs:
+            if self.token_index == 6:
+                for key, value in attrs:
+                    if key == 'value':
+                        APP_UPDATE_AUTH_TOKEN = value
+                self.token_index += 1
+            else:
+                self.token_index += 1
+
+
 class ConfigCache(object):
     def __init__(self, zk):
         self._load(**zk.list_configuration())
         self._refresh_zk(zk)
 
+
+
     def _load(self, **kwargs):
         for attr, value in kwargs.items():
             if not hasattr(self, attr):
                 if not value:
                     value = CONFIG_LOCAL.get(attr)
+                else:
+                    if attr in ['github_user_passwd', 'dns_provider_token',
+                                'github_user_token']:
```

Comment:
it's good set a variable for the list, like `base64_encoded_options` so that we can update it easily in the future just in one place.

Reply:
done